benchmarks/segmentation_layers.py

```diff
@@ -1,175 +1,319 @@
+            for o in obj:
+                res = self.get_max_size(o, res)
+
+        if isinstance(obj, nestedtensor.nested.nested.NestedTensor):
+            tres = self.get_max_size(obj.unbind())
+            while len(tres) > len(res):
+                    res.append(0)
+
+            res = [max(i, j) for (i, j) in zip(res, tres)]
+
+        if isinstance(obj, torch.Tensor):
+            # scalar
+            if obj.dim() == 0 and obj.numel() == 1:
+                res = [1]
+            else:
+                while len(obj.size()) > len(res):
+                    res.append(0)
+
+                res = [max(i, j) for (i, j) in zip(res, obj.size())]
+
+        return res
+
+    def pad_tensor_to_shape(self, t, goal_shape):
+        padd = ()
+        tup = tuple(t.size())
+        assert(t.dim() == len(goal_shape))
+        for i in range(len(tup)):
+            padd = (0, goal_shape[i] - tup[i]) + padd
+        new_tensor = F.pad(t, padd)
+        new_tensor = new_tensor.reshape(goal_shape)
+        return new_tensor
+
+    def pad_tensors_to_max_shape(self, inputs):
+        max_shape = self.get_max_size(inputs)
+        padded_inputs = [self.pad_tensor_to_shape(t, max_shape) for t in inputs]
+        return padded_inputs
+
+    def get_input(self, return_targets=False, pad=False):
+        inputs = []
+        targets = []
+        
+        torch.manual_seed(self.seed)
+        for i in range(self.N):
+            h_delta = random.gauss(self.H, self.var_h)
+            w_delta = random.gauss(self.H, self.var_w)
+            h = int(self.H + h_delta)
+            w = int(self.W + w_delta)
+            inputs.append(torch.randn(self.C, h, w))
+            targets.append(torch.randint(1, (h, w), dtype=torch.int64))
+
+        if pad:
+            inputs = self.pad_tensors_to_max_shape(inputs)
+
+        if return_targets:
+            return inputs, targets
+
+        return inputs
+
+    #
+    # relu
+    #
+    def relu_tensor_iter(self):
+        inputs = self.get_input()
+
+        def _relu_tensor_iter():
+            for t in inputs:
+                torch.nn.functional.relu(t)
+        return _relu_tensor_iter
+
+    def relu_tensor_pad(self):
+        inputs = self.get_input(pad=True)
+        stack = torch.stack(inputs)
+
+        def _relu_tensor_pad():
+            torch.nn.functional.relu(stack)
+
+        return _relu_tensor_pad
+
+    def relu_nt(self):
+        inputs = self.get_input()
+        nt = nestedtensor.nested_tensor(inputs)
+        def _relu_nt():
+            torch.nn.functional.relu(nt)
+        
+        return _relu_nt
+
+    #
+    # conv2d
+    #
+    def conv2d_tensor_iter(self):
+        inputs = self.get_input()
+
+        def _conv2d_tensor_iter():
+            for t in inputs:
+                self.conv2d(t.unsqueeze(0)).squeeze(0)
+        return _conv2d_tensor_iter
+
+    def conv2d_tensor_pad(self):
+        inputs = self.get_input(pad=True)
```

Comment:
Patterns such as 
```
        inputs = self.get_input(pad=True)
        stack = torch.stack(inputs)
```

or

```
inputs = self.get_input()
```

from  ```conv2d_tensor_iter``` are repeated across the other benchmarks. I think you can abstract those and generate the functions instead. That might save you a few dozen lines.